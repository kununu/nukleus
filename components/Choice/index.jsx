import React, {Component, PropTypes} from 'react';

import styles from './index.scss';

import {
  controlLabel,
  controlLabelRequired,
  controlNote,
  formGroup
} from '../index.scss';


export default class Choice extends Component {
  static propTypes = {
    checked: PropTypes.string,
    customTheme: PropTypes.string,
    disabled: PropTypes.bool,
    heading: PropTypes.string,
    headingStyle: PropTypes.string,
    isRequired: PropTypes.bool,
    name: PropTypes.string.isRequired,
    onChange: PropTypes.func,
    options: PropTypes.array.isRequired,
    query: PropTypes.object,
    requiredLabel: PropTypes.string
  };

  static defaultProps = {
    checked: '',
    customTheme: '',
    disabled: false,
    heading: '',
    headingStyle: 'control-label',
    isRequired: false,
    onChange: null,
    query: {},
    requiredLabel: ''
  };

  state = {
    checked: this.props.checked
  };

  componentWillMount () {
    const {query, name} = this.props;
    if (!query[name]) return;
    this.setState({
      checked: query[name]
    });
  }

  componentWillReceiveProps (nextProps) {
    const {query, name} = nextProps;
    if (nextProps.query === this.props.query && nextProps.checked === this.props.checked) return;
    if (query[name] && query[name] !== this.props.query[name]) {
      this.setState({
        checked: query[name]
      });
    } else if (nextProps.checked !== this.props.checked) {
      this.setState({
        checked: nextProps.checked
      });
    }
  }

  onChange = e => {
    if (this.props.disabled) return;

    this.props.onChange(e);
    this.setState({
      checked: e.target.value
    });
  };

  render () {
    const {
      customTheme,
      disabled,
      heading,
      isRequired,
      name,
      options,
      requiredLabel
    } = this.props;

    const {
      checked
    } = this.state;

    return (
      <div className={formGroup}>

        {requiredLabel &&
          <span className={`${controlNote} ${controlLabelRequired}`}>
            {requiredLabel}
          </span>
        }

        {heading && <div className={`${this.props.headingStyle} ${controlLabel}`}>{heading}</div>}

        <div className={`${styles.radioContainer} ${options.length > 3 && styles.flexible}`}>
          {options.map((item, idx) =>
            <div className={styles.radioButton} key={item.id}>
              <input
                type="radio"
                value={item.value}
                id={`${name}${item.id}`}
                name={name}
                checked={checked === item.value}
                onChange={this.onChange}
                required={isRequired} />
              <label
                disabled={disabled}
                id={idx}
                htmlFor={`${name}${item.id}`}
                className={customTheme}>
                {item.label}
              </label>
            </div>
          )}
        </div>
      </div>
    );
  }
}
