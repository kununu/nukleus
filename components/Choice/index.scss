@import '~kununu-theme/scss/base/variables';

.radioContainer {
  display: table;
  width: 100%;

  * {
    box-sizing: border-box;
  }

  @supports (display: flex) {
    display: flex;
    flex-flow: row wrap;
  }
}

.radioButton {
  text-align: center;

  @supports (display: flex) {
    display: block;
    flex: 1;
  }

  input {
    position: absolute;
    opacity: 0;
    z-index: -1;
  }

  input:checked + label {
    &,
    &:hover {
      background: $blue-light;
      color: $white;
    }
  }

  label {
    width: 100%;
    height: 100%;
    margin: 0;
    padding: 11px 15px;
    cursor: pointer;
    font-weight: normal;
    transition: all .25s;
    border: 2px solid $blue-light;
    border-left: 0;
    color: $blue-dark;

    @supports (display: flex) {
      display: flex;
      justify-content: center;
      align-items: center;
    }

    &:hover {
      background: $gray-lightest;
    }

    &.disabled,
    &[disabled] {
      cursor: not-allowed;
      opacity: .65;
      border-color: $blue-light;

      &:hover {
        background: initial;
      }
    }
  }

  &:first-child {
    label {
      border-radius: $border-radius-base 0 0 $border-radius-base;
      border-left: 2px solid $blue-light;
    }
  }

  &:last-child {
    label {
      border-radius: 0 $border-radius-base $border-radius-base 0;
    }
  }
}

.radioContainer.flexible {
  @media (max-width: $screen-sm-max) {
    border: 2px solid $blue;
    border-radius: $border-radius-base;

    .radioButton {
      flex: 50%;

      label {
        border: 0;
        border-radius: 0;
      }

      &:nth-child(4n+3) label,
      &:nth-child(4n+4) label {
        border: 2px solid $blue;
        border-radius: 0;
        border-left: 0;
        border-right: 0;
      }

      &:nth-child(odd) label {
        border-right: 2px solid $blue;
      }

      &:nth-last-child(2) label,
      &:nth-last-child(1) label {
        border-bottom: 0;
      }
    }
  }
}
