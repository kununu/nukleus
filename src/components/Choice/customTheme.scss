@import '~@kununu/kununu-theme-v2/scss/base/variables';

.choiceButton {
  text-align: center;
  min-height: 20px;
  display: flex;
  flex: 1;

  input {
    position: absolute;
    opacity: 0;
    z-index: -1;
  }

  input:checked + label {
    &,
    &:hover {
      background: $green;
      color: $white;
    }
  }

  label {
    width: 100%;
    margin: 10px;
    padding: 11px 15px;
    cursor: pointer;
    font-weight: $font-weight-semibold;
    transition: all .25s;
    text-transform: uppercase;
    border: 1px solid $green;
    border-radius: 5px;
    color: $green;
    display: flex;
    justify-content: center;
    align-items: center;

    &:hover {
      background: $gray-lightest;
    }

    &.disabled,
    &[disabled] {
      cursor: not-allowed;
      opacity: .65;

      &:hover {
        background: initial;
      }
    }
  }
}