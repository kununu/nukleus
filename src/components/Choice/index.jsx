import React from 'react';
import PropTypes from 'prop-types';

import {queryParamsToObject} from 'utils/params';


import Error from '../Error';
import Label from '../Label';
import {
  controlLabelRequired,
  controlNote,
  formGroup,
} from '../index.scss';

import styles from './index.scss';

export default class Choice extends React.Component {
  static propTypes = {
    checked: PropTypes.string,
    customTheme: PropTypes.string,
    disabled: PropTypes.bool,
    error: PropTypes.string,
    errorSubInfo: PropTypes.string,
    heading: PropTypes.string,
    headingStyle: PropTypes.string,
    isRequired: PropTypes.bool,
    label: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.object,
    ]),
    labelHidden: PropTypes.bool,
    name: PropTypes.string.isRequired,
    onBlur: PropTypes.func,
    onChange: PropTypes.func,
    onClick: PropTypes.func,
    onFocus: PropTypes.func,
    options: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.string,
      label: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.object,
      ]),
      value: PropTypes.string,
    })).isRequired,
    optionsPerRow: PropTypes.oneOf(['3', '4', '5', '6', '7', 3, 4, 5, 6, 7, null]),
    query: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.object,
    ]),
    reference: PropTypes.func,
    requiredLabel: PropTypes.string,
  };

  static defaultProps = {
    checked: '',
    customTheme: '',
    disabled: false,
    error: null,
    errorSubInfo: null,
    heading: null,
    headingStyle: 'control-label',
    isRequired: false,
    label: null,
    labelHidden: false,
    onBlur: () => {},
    onChange: () => {},
    onClick: () => {},
    onFocus: () => {},
    optionsPerRow: null,
    query: {},
    reference: () => {},
    requiredLabel: '',
  };

  state = {
    checked: this.props.checked, // eslint-disable-line react/destructuring-assignment
    showError: false,
  };

  componentWillMount () {
    const {
      error,
      name,
      query,
    } = this.props;
    const queryObject = queryParamsToObject(query);

    // Show error, if already set
    if (error !== null) this.showError();

    if (!queryObject[name]) return;

    this.setState({checked: queryObject[name]});
  }

  componentWillReceiveProps (nextProps) {
    const {
      checked,
      error,
      name,
      query,
    } = this.props;
    const queryNextObject = queryParamsToObject(nextProps.query);
    const queryPropsObject = queryParamsToObject(query);

    if (error) this.showError();
    if (nextProps.query === query && nextProps.checked === checked) return;

    if (queryNextObject[name] && queryNextObject[name] !== queryPropsObject[name]) {
      this.setState({checked: queryNextObject[name]});
    } else if (nextProps.checked !== checked) {
      this.setState({checked: nextProps.checked});
    }
  }

  onChange = (option, e) => {
    const {value} = e.target;
    const {checked} = this.state;
    const {onChange} = this.props;

    if (this.isOptionDisabled(option) || value === checked) return;

    onChange(e);
    this.setState({checked: value});
  };

  onClick = (option, e) => {
    if (this.isOptionDisabled(option)) return;

    const {value} = e.target;
    const {checked} = this.state;
    const {
      onClick,
      isRequired,
    } = this.props;

    onClick(e);
    // As long as the component is not required and the component is deselected set to null.
    if (!isRequired && value === checked) {
      this.setState({checked: null});
    }
  }

  get label () {
    const {
      heading,
      headingStyle,
      label,
      labelHidden,
    } = this.props;

    if (!label && !heading) return null;

    const value = label || heading;

    return (
      <Label
        htmlFor={heading}
        value={value}
        labelHidden={labelHidden}
        classNames={headingStyle}
        isTitle
      />
    );
  }

  isOptionDisabled (option) {
    const {disabled} = this.props;

    return disabled || (typeof (option.disabled) === 'boolean' ? option.disabled : disabled);
  }

  hasError () {
    const {error} = this.props;
    const {showError} = this.state;

    return showError && error;
  }

  hideError () {
    this.setState({showError: false});
  }

  showError () {
    this.setState({showError: true});
  }

  render () {
    const {
      customTheme,
      error,
      errorSubInfo,
      isRequired,
      name,
      onBlur,
      onFocus,
      options,
      optionsPerRow,
      reference,
      requiredLabel,
    } = this.props;
    const {checked} = this.state;

    return (
      <div
        className={formGroup}
        id={`${name}-container`}
      >

        {requiredLabel && (
        <span className={`${controlNote} ${controlLabelRequired}`}>
          {requiredLabel}
        </span>
        )}

        {this.label}

        <div
          className={`${styles.radioContainer} ${options.length > 3 && optionsPerRow === null && styles.flexible}`}
          data-options-per-row={optionsPerRow}
        >
          {options.map((item, idx) => (
            <div
              className={styles.radioButton}
              key={item.id}
            >
              <input
                type="radio"
                value={item.value}
                id={`${name}${item.id}`}
                name={name}
                checked={checked === item.value}
                onBlur={onBlur}
                onChange={e => this.onChange(item, e)}
                onFocus={onFocus}
                onClick={e => this.onClick(item, e)}
                ref={reference}
                required={isRequired}
              />
              <label
                disabled={this.isOptionDisabled(item)}
                id={idx}
                htmlFor={`${name}${item.id}`}
                className={customTheme}
              >
                {item.label}
              </label>
            </div>
          ))}
        </div>

        {this.hasError() && (
        <Error
          info={error}
          subInfo={errorSubInfo}
        />
        )}
      </div>
    );
  }
}
