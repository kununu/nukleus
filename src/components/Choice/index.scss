@import '~@kununu/kununu-theme-v2/scss/base/variables';

.choiceContainer {
  width: 100%;
  display: flex;
  flex-flow: row wrap;
  flex-wrap: wrap;

  * {
    box-sizing: border-box;
  }
}

.choiceButton {
  text-align: center;
  min-height: 50px;
  display: flex;
  flex: 1;

  input {
    position: absolute;
    opacity: 0;
    z-index: -1;
  }

  input:checked + label {
    &,
    &:hover {
      background: $blue;
      color: $white;
    }
  }

  label {
    width: 100%;
    margin: 0;
    padding: 11px 15px;
    cursor: pointer;
    font-weight: $font-weight-semibold;
    transition: all .25s;
    border: 1px solid $blue;
    border-left: 0;
    border-top: 0;
    color: $blue;
    display: flex;
    justify-content: center;
    align-items: center;

    &:hover {
      background: $gray-lightest;
    }

    &.disabled,
    &[disabled] {
      cursor: not-allowed;
      opacity: .65;
      border-color: $blue-light;

      &:hover {
        background: initial;
      }
    }
  }

  &:first-child {
    label {
      border-radius: 0;
      border-left: 1px solid $blue;
    }
  }

  &:last-child {
    label {
      border-radius: 0;
    }
  }

  @media (max-width: $screen-sm-max) {
    flex: 0 0 50%;

    // First Row
    &:nth-child(-n+2) {
      label {
        border-top: 1px solid $blue;
      }
    }

    // All Left Radio Buttons
    &:nth-child(2n + 1) {
      label {
        border-left: 1px solid $blue;
      }
    }

    // Style for border corners
    // Top Left Corner
    &:first-child {
      label {
        border-top-left-radius: $border-radius-base;
      }
    }

    // Top Right Corner
    &:nth-child(2) {
      label {
        border-top-right-radius: $border-radius-base;
      }
    }

    // Bottom Left Corner
    &:nth-child(2n + 1):nth-last-child(-n + 2) {
      label {
        border-bottom-left-radius: $border-radius-base;
      }
    }

    // Bottom Right Corner
    &:last-child,
    &:nth-child(2n):nth-last-child(-n + 2) {
      label {
        border-bottom-right-radius: $border-radius-base;
      }
    }
  }
}

.choiceFlexible {
  @media (max-width: $screen-sm-max) {
    border: 1px solid $blue;
    border-radius: $border-radius-base;

    .choiceButton {
      display: flex;
      flex: 50%;

      label {
        border: 0;
        border-radius: 0;
      }

      &:nth-child(4n+3) label,
      &:nth-child(4n+4) label {
        border: 1px solid $blue;
        border-radius: 0;
        border-left: 0;
        border-right: 0;
      }

      &:nth-child(odd) label {
        border-right: 1px solid $blue;
      }

      &:nth-last-child(2) label,
      &:nth-last-child(1) label {
        border-bottom: 0;
      }
    }
  }
}

@mixin options-per-row($count) {
  [data-options-per-row='#{$count}'] {
    display: flex;
    
    @supports(display: grid) {
      display: grid;
      grid-template-columns: 1fr 1fr;
      grid-auto-rows: 1fr;

      @media (min-width: $screen-md-min) {
        grid-template-columns: repeat($count, 1fr);
      }
    }

    @media (min-width: $screen-md-min) {
      .choiceButton {
        flex: 0 0 100% / $count;

        label {
          border-top: 0;
        }
      }

      // Target first row
      .choiceButton:nth-child(-n + #{$count}) {
        label {
          border-top: 1px solid $blue;
        }
      }

      // All left choiceButtons
      .choiceButton:nth-child(#{$count}n + 1) {
        label {
          border-left: 1px solid $blue;
        }
      }

      // Top Left Corner
      .choiceButton:first-child {
        label {
          border-top-left-radius: $border-radius-base;
        }
      }

      // Top Right Corner
      .choiceButton:nth-child(#{$count}) {
        label {
          border-top-right-radius: $border-radius-base;
        }
      }

      // Bottom Left Corner
      .choiceButton:nth-child(#{$count}n+1):nth-last-child(-n+#{$count}) {
        label {
          border-bottom-left-radius: $border-radius-base;
        }
      }

      // Bottom Right Corner
      .choiceButton:nth-child(#{$count}n):nth-last-child(-n + #{$count}),
      .choiceButton:last-child {
        label {
          border-bottom-right-radius: $border-radius-base;
        }
      }
    }
  }
}

@include options-per-row(3);
@include options-per-row(4);
@include options-per-row(5);
@include options-per-row(6);
@include options-per-row(7);
