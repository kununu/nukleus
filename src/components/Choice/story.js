import React from 'react';
import {storiesOf} from '@storybook/react';
import {defaultOptions} from 'defaultOptions';
import {withOptions} from '@storybook/addon-options';
// import {action} from '@storybook/addon-actions';
// import {withNotes} from '@storybook/addon-notes';
// import {withInfo} from '@storybook/addon-info';
// import {withKnobs, text, boolean} from '@storybook/addon-knobs/react';
// import backgrounds from '@storybook/addon-backgrounds';

import Choice from './index';

storiesOf('Choice', module)
  .addDecorator(withOptions({...defaultOptions}))
  .add('with some emoji', () => (
    <Choice
      label="Choice"
      name="basic"
      onChange={() => {}}
      checked="option-a"
      options={[
        {
          id: 'option-a',
          label: 'Option A',
          value: 'option-a'
        },
        {
          id: 'option-b',
          label: 'Option B',
          value: 'option-b'
        },
        {
          id: 'option-c',
          label: 'Option C',
          value: 'option-c'
        }
      ]}
    // {/>}
  ));
