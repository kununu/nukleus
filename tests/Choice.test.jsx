import React from 'react';
import renderer from 'react-test-renderer';
import {mount} from 'enzyme';
import Choice from 'Choice'; // eslint-disable-line import/no-unresolved, import/extensions, import/no-extraneous-dependencies

const options = [
  {
    id: 'opA',
    label: 'Option A',
    value: 'opA'
  },
  {
    id: 'opB',
    label: 'Option B',
    value: 'opB'
  },
  {
    id: 'opC',
    label: 'Option C',
    value: 'opC'
  }
];

test('Renders Choice with radio buttons without crashing', () => {
  const component = renderer.create(
    <Choice
      name="test"
      options={options}
      onChange={() => {}} />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders Choice with an heading without crashing', () => {
  const component = renderer.create(
    <Choice
      heading="heading"
      name="test"
      options={options}
      onChange={() => {}} />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('Renders a disabled Choice without crashing', () => {
  const component = renderer.create(
    <Choice
      name="test"
      options={options}
      disabled
      onChange={() => {}} />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders a Choice with default checked without crashing', () => {
  const component = renderer.create(
    <Choice
      name="test"
      options={options}
      checked="opB"
      onChange={() => {}} />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders a Choice with no default checked if it\'s not an option', () => {
  const component = renderer.create(
    <Choice
      name="test"
      options={options}
      checked="opD"
      onChange={() => {}} />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders a Choice with checked from route', () => {
  const component = renderer.create(
    <Choice
      name="test"
      options={options}
      onChange={() => {}}
      query={{test: 'opC'}} />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders a Choice with no default checked from route if it is not an option', () => {
  const component = renderer.create(
    <Choice
      name="test"
      options={options}
      onChange={() => {}}
      query={{test: 'opD'}} />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders a Choice with no default checked from route if it is the other name', () => {
  const component = renderer.create(
    <Choice
      name="test"
      options={options}
      onChange={() => {}}
      query={{otherTest: 'opC'}} />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders a Choice with custom style', () => {
  const component = renderer.create(
    <Choice
      name="test"
      options={options}
      customTheme="customThemeClass"
      onChange={() => {}} />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders a Choice with custom style disabled', () => {
  const component = renderer.create(
    <Choice
      name="test"
      options={options}
      customTheme="customThemeClass"
      disabled
      onChange={() => {}} />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Change checked when changed', () => {
  const component = mount(
    <Choice
      name="test"
      options={options}
      onChange={() => {}} />
  );

  expect(component.state('checked')).toEqual('');
  component.find('#testopA').simulate('change');
  expect(component.state('checked')).toEqual('opA');
});

test('Fires onChange function when changed', () => {
  const spyFunc = jest.fn();
  const component = mount(
    <Choice
      name="test"
      options={options}
      onChange={spyFunc} />
  );

  component.find('#testopA').simulate('change');
  expect(spyFunc).toHaveBeenCalled();
});

test('Does not fire onChange function when clicked if Choice is disabled', () => {
  const spyFunc = jest.fn();
  const component = mount(
    <Choice
      name="test"
      options={options}
      disabled
      onChange={spyFunc} />
  );

  component.find('#testopA').simulate('change');
  expect(spyFunc).not.toHaveBeenCalled();
});

test('Change checked prop change checked state', () => {
  const component = mount(
    <Choice
      name="test"
      checked="opA"
      options={options}
      onChange={() => {}} />
  );

  expect(component.state('checked')).toEqual('opA');
  component.setProps({checked: 'opB'});
  expect(component.state('checked')).toEqual('opB');
});

test('No change checked prop does not change checked state', () => {
  const component = mount(
    <Choice
      name="test"
      checked="opA"
      options={options}
      onChange={() => {}} />
  );

  component.setState({checked: 'opB'});
  component.setProps({checked: 'opA'});
  expect(component.state('checked')).toEqual('opB');
});


test('Change checked query changes state', () => {
  const component = mount(
    <Choice
      name="test"
      checked="opA"
      options={options}
      onChange={() => {}} />
  );

  expect(component.state('checked')).toEqual('opA');
  component.setProps({query: {test: 'opB'}});
  expect(component.state('checked')).toEqual('opB');
});

test('No change checked query does not change state', () => {
  const component = mount(
    <Choice
      name="test"
      query={{test: 'opA'}}
      options={options}
      onChange={() => {}} />
  );

  component.setState({checked: 'opB'});
  expect(component.state('checked')).toEqual('opB');
  component.setProps({query: {test: 'opA'}});
  expect(component.state('checked')).toEqual('opB');
});
