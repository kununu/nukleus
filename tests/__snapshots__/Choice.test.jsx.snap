exports[`test Renders Choice with an heading without crashing 1`] = `
<div
  className="formGroup">
  
  <div
    className="control-label controlLabel">
    heading
  </div>
  <div
    className="radioContainer false"
    data-options-per-row={null}>
    <div
      className="radioButton">
      <input
        checked={false}
        id="testopA"
        name="test"
        onChange={[Function]}
        required={false}
        type="radio"
        value="opA" />
      <label
        className=""
        disabled={false}
        htmlFor="testopA"
        id={0}>
        Option A
      </label>
    </div>
    <div
      className="radioButton">
      <input
        checked={false}
        id="testopB"
        name="test"
        onChange={[Function]}
        required={false}
        type="radio"
        value="opB" />
      <label
        className=""
        disabled={false}
        htmlFor="testopB"
        id={1}>
        Option B
      </label>
    </div>
    <div
      className="radioButton">
      <input
        checked={false}
        id="testopC"
        name="test"
        onChange={[Function]}
        required={false}
        type="radio"
        value="opC" />
      <label
        className=""
        disabled={false}
        htmlFor="testopC"
        id={2}>
        Option C
      </label>
    </div>
  </div>
</div>
`;

exports[`test Renders Choice with radio buttons without crashing 1`] = `
<div
  className="formGroup">
  
  
  <div
    className="radioContainer false"
    data-options-per-row={null}>
    <div
      className="radioButton">
      <input
        checked={false}
        id="testopA"
        name="test"
        onChange={[Function]}
        required={false}
        type="radio"
        value="opA" />
      <label
        className=""
        disabled={false}
        htmlFor="testopA"
        id={0}>
        Option A
      </label>
    </div>
    <div
      className="radioButton">
      <input
        checked={false}
        id="testopB"
        name="test"
        onChange={[Function]}
        required={false}
        type="radio"
        value="opB" />
      <label
        className=""
        disabled={false}
        htmlFor="testopB"
        id={1}>
        Option B
      </label>
    </div>
    <div
      className="radioButton">
      <input
        checked={false}
        id="testopC"
        name="test"
        onChange={[Function]}
        required={false}
        type="radio"
        value="opC" />
      <label
        className=""
        disabled={false}
        htmlFor="testopC"
        id={2}>
        Option C
      </label>
    </div>
  </div>
</div>
`;

exports[`test Renders a Choice with checked from route 1`] = `
<div
  className="formGroup">
  
  
  <div
    className="radioContainer false"
    data-options-per-row={null}>
    <div
      className="radioButton">
      <input
        checked={false}
        id="testopA"
        name="test"
        onChange={[Function]}
        required={false}
        type="radio"
        value="opA" />
      <label
        className=""
        disabled={false}
        htmlFor="testopA"
        id={0}>
        Option A
      </label>
    </div>
    <div
      className="radioButton">
      <input
        checked={false}
        id="testopB"
        name="test"
        onChange={[Function]}
        required={false}
        type="radio"
        value="opB" />
      <label
        className=""
        disabled={false}
        htmlFor="testopB"
        id={1}>
        Option B
      </label>
    </div>
    <div
      className="radioButton">
      <input
        checked={true}
        id="testopC"
        name="test"
        onChange={[Function]}
        required={false}
        type="radio"
        value="opC" />
      <label
        className=""
        disabled={false}
        htmlFor="testopC"
        id={2}>
        Option C
      </label>
    </div>
  </div>
</div>
`;

exports[`test Renders a Choice with custom style 1`] = `
<div
  className="formGroup">
  
  
  <div
    className="radioContainer false"
    data-options-per-row={null}>
    <div
      className="radioButton">
      <input
        checked={false}
        id="testopA"
        name="test"
        onChange={[Function]}
        required={false}
        type="radio"
        value="opA" />
      <label
        className="customThemeClass"
        disabled={false}
        htmlFor="testopA"
        id={0}>
        Option A
      </label>
    </div>
    <div
      className="radioButton">
      <input
        checked={false}
        id="testopB"
        name="test"
        onChange={[Function]}
        required={false}
        type="radio"
        value="opB" />
      <label
        className="customThemeClass"
        disabled={false}
        htmlFor="testopB"
        id={1}>
        Option B
      </label>
    </div>
    <div
      className="radioButton">
      <input
        checked={false}
        id="testopC"
        name="test"
        onChange={[Function]}
        required={false}
        type="radio"
        value="opC" />
      <label
        className="customThemeClass"
        disabled={false}
        htmlFor="testopC"
        id={2}>
        Option C
      </label>
    </div>
  </div>
</div>
`;

exports[`test Renders a Choice with custom style disabled 1`] = `
<div
  className="formGroup">
  
  
  <div
    className="radioContainer false"
    data-options-per-row={null}>
    <div
      className="radioButton">
      <input
        checked={false}
        id="testopA"
        name="test"
        onChange={[Function]}
        required={false}
        type="radio"
        value="opA" />
      <label
        className="customThemeClass"
        disabled={true}
        htmlFor="testopA"
        id={0}>
        Option A
      </label>
    </div>
    <div
      className="radioButton">
      <input
        checked={false}
        id="testopB"
        name="test"
        onChange={[Function]}
        required={false}
        type="radio"
        value="opB" />
      <label
        className="customThemeClass"
        disabled={true}
        htmlFor="testopB"
        id={1}>
        Option B
      </label>
    </div>
    <div
      className="radioButton">
      <input
        checked={false}
        id="testopC"
        name="test"
        onChange={[Function]}
        required={false}
        type="radio"
        value="opC" />
      <label
        className="customThemeClass"
        disabled={true}
        htmlFor="testopC"
        id={2}>
        Option C
      </label>
    </div>
  </div>
</div>
`;

exports[`test Renders a Choice with default checked without crashing 1`] = `
<div
  className="formGroup">
  
  
  <div
    className="radioContainer false"
    data-options-per-row={null}>
    <div
      className="radioButton">
      <input
        checked={false}
        id="testopA"
        name="test"
        onChange={[Function]}
        required={false}
        type="radio"
        value="opA" />
      <label
        className=""
        disabled={false}
        htmlFor="testopA"
        id={0}>
        Option A
      </label>
    </div>
    <div
      className="radioButton">
      <input
        checked={true}
        id="testopB"
        name="test"
        onChange={[Function]}
        required={false}
        type="radio"
        value="opB" />
      <label
        className=""
        disabled={false}
        htmlFor="testopB"
        id={1}>
        Option B
      </label>
    </div>
    <div
      className="radioButton">
      <input
        checked={false}
        id="testopC"
        name="test"
        onChange={[Function]}
        required={false}
        type="radio"
        value="opC" />
      <label
        className=""
        disabled={false}
        htmlFor="testopC"
        id={2}>
        Option C
      </label>
    </div>
  </div>
</div>
`;

exports[`test Renders a Choice with no default checked from route if it is not an option 1`] = `
<div
  className="formGroup">
  
  
  <div
    className="radioContainer false"
    data-options-per-row={null}>
    <div
      className="radioButton">
      <input
        checked={false}
        id="testopA"
        name="test"
        onChange={[Function]}
        required={false}
        type="radio"
        value="opA" />
      <label
        className=""
        disabled={false}
        htmlFor="testopA"
        id={0}>
        Option A
      </label>
    </div>
    <div
      className="radioButton">
      <input
        checked={false}
        id="testopB"
        name="test"
        onChange={[Function]}
        required={false}
        type="radio"
        value="opB" />
      <label
        className=""
        disabled={false}
        htmlFor="testopB"
        id={1}>
        Option B
      </label>
    </div>
    <div
      className="radioButton">
      <input
        checked={false}
        id="testopC"
        name="test"
        onChange={[Function]}
        required={false}
        type="radio"
        value="opC" />
      <label
        className=""
        disabled={false}
        htmlFor="testopC"
        id={2}>
        Option C
      </label>
    </div>
  </div>
</div>
`;

exports[`test Renders a Choice with no default checked from route if it is the other name 1`] = `
<div
  className="formGroup">
  
  
  <div
    className="radioContainer false"
    data-options-per-row={null}>
    <div
      className="radioButton">
      <input
        checked={false}
        id="testopA"
        name="test"
        onChange={[Function]}
        required={false}
        type="radio"
        value="opA" />
      <label
        className=""
        disabled={false}
        htmlFor="testopA"
        id={0}>
        Option A
      </label>
    </div>
    <div
      className="radioButton">
      <input
        checked={false}
        id="testopB"
        name="test"
        onChange={[Function]}
        required={false}
        type="radio"
        value="opB" />
      <label
        className=""
        disabled={false}
        htmlFor="testopB"
        id={1}>
        Option B
      </label>
    </div>
    <div
      className="radioButton">
      <input
        checked={false}
        id="testopC"
        name="test"
        onChange={[Function]}
        required={false}
        type="radio"
        value="opC" />
      <label
        className=""
        disabled={false}
        htmlFor="testopC"
        id={2}>
        Option C
      </label>
    </div>
  </div>
</div>
`;

exports[`test Renders a Choice with no default checked if it's not an option 1`] = `
<div
  className="formGroup">
  
  
  <div
    className="radioContainer false"
    data-options-per-row={null}>
    <div
      className="radioButton">
      <input
        checked={false}
        id="testopA"
        name="test"
        onChange={[Function]}
        required={false}
        type="radio"
        value="opA" />
      <label
        className=""
        disabled={false}
        htmlFor="testopA"
        id={0}>
        Option A
      </label>
    </div>
    <div
      className="radioButton">
      <input
        checked={false}
        id="testopB"
        name="test"
        onChange={[Function]}
        required={false}
        type="radio"
        value="opB" />
      <label
        className=""
        disabled={false}
        htmlFor="testopB"
        id={1}>
        Option B
      </label>
    </div>
    <div
      className="radioButton">
      <input
        checked={false}
        id="testopC"
        name="test"
        onChange={[Function]}
        required={false}
        type="radio"
        value="opC" />
      <label
        className=""
        disabled={false}
        htmlFor="testopC"
        id={2}>
        Option C
      </label>
    </div>
  </div>
</div>
`;

exports[`test Renders a Choice with optionsPerRow set without crashing 1`] = `
<div
  className="formGroup">
  
  
  <div
    className="radioContainer false"
    data-options-per-row={3}>
    <div
      className="radioButton">
      <input
        checked={false}
        id="testopA"
        name="test"
        onChange={[Function]}
        required={false}
        type="radio"
        value="opA" />
      <label
        className=""
        disabled={false}
        htmlFor="testopA"
        id={0}>
        Option A
      </label>
    </div>
    <div
      className="radioButton">
      <input
        checked={false}
        id="testopB"
        name="test"
        onChange={[Function]}
        required={false}
        type="radio"
        value="opB" />
      <label
        className=""
        disabled={false}
        htmlFor="testopB"
        id={1}>
        Option B
      </label>
    </div>
    <div
      className="radioButton">
      <input
        checked={false}
        id="testopC"
        name="test"
        onChange={[Function]}
        required={false}
        type="radio"
        value="opC" />
      <label
        className=""
        disabled={false}
        htmlFor="testopC"
        id={2}>
        Option C
      </label>
    </div>
  </div>
</div>
`;

exports[`test Renders a disabled Choice without crashing 1`] = `
<div
  className="formGroup">
  
  
  <div
    className="radioContainer false"
    data-options-per-row={null}>
    <div
      className="radioButton">
      <input
        checked={false}
        id="testopA"
        name="test"
        onChange={[Function]}
        required={false}
        type="radio"
        value="opA" />
      <label
        className=""
        disabled={true}
        htmlFor="testopA"
        id={0}>
        Option A
      </label>
    </div>
    <div
      className="radioButton">
      <input
        checked={false}
        id="testopB"
        name="test"
        onChange={[Function]}
        required={false}
        type="radio"
        value="opB" />
      <label
        className=""
        disabled={true}
        htmlFor="testopB"
        id={1}>
        Option B
      </label>
    </div>
    <div
      className="radioButton">
      <input
        checked={false}
        id="testopC"
        name="test"
        onChange={[Function]}
        required={false}
        type="radio"
        value="opC" />
      <label
        className=""
        disabled={true}
        htmlFor="testopC"
        id={2}>
        Option C
      </label>
    </div>
  </div>
</div>
`;
